projects/igniteui-angular/src/lib/calendar/calendar.component.ts

```diff
@@ -1089,13 +1089,14 @@ export class IgxCalendarComponent implements OnInit, ControlValueAccessor {
      */
     private selectMultiple(value: Date | Date[]) {
```

Comment:
The selectedDates are the calendar value. If we change the setting of selection to use DateRangeDescriptor  the user need to pass selectedDates with DateRangeDescriptor and the value also. In case with single selection a DateRangeDescriptor  with only one date need to be passed.